src/augeas.c

```diff
@@ -1968,6 +1968,77 @@ int aug_source(const augeas *aug, const char *path, char **file_path) {
     return result;
 }
 
+int aug_preview(struct augeas *aug, const char *path, char **out) {
+    struct tree *tree = NULL;
+    struct pathx *p;
+    int r;
+    char *lens_path = NULL;
+    char *lens_name = NULL;
+    char *file_path = NULL;
+    char *source_filename = NULL;
+    char *source_text = NULL;
+
+    *out = NULL;
+
+    api_entry(aug);
+
+    p = pathx_aug_parse(aug, aug->origin, tree_root_ctx(aug), path, true);
+    ERR_BAIL(aug);
+
+    tree = pathx_first(p);
+    ERR_BAIL(aug);
+    ERR_THROW(tree == NULL, aug, AUG_ENOMATCH, "No node matching %s", path);
+
+    file_path = tree_source(aug, tree);
+
+    ERR_THROW(file_path == NULL, aug, AUG_EBADARG, "Path %s is not associated with a file", path);
+
+    if ( file_path == NULL ) {
+        file_path = strdup(path);
+        ERR_NOMEM(file_path == NULL, aug);
+    } else {
+        tree = tree_find(aug, file_path);
+    }
+
+    r = xasprintf(&lens_path, "%s%s/%s", AUGEAS_META_TREE, file_path, s_lens);
+    if ( r>= 0 ) {
```

Comment:
This is super nitpicky, but I generally write `if` statements with now spaces after the parentheses, but spaces around the operator, i.e., `if (r >= 0) { ..` - I wish there was some easy-to-use code formatter that we could just run to make sure these things are consistent. In any event, I would prefer to have these things unfirom as much as possible, but it's a preference, not a "I would never merge a PR formatted like this" ;)

Reply:
Instead of the `if` statement, it would be better to write
```
ERR_NOMEM(lens_path == NULL, aug)
```

Reply:
Replaced `if (r >= 0)` with `ERR_NOMEM` macro